TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceResource.robot

```diff
@@ -0,0 +1,130 @@
+
+*** Test Cases ***
+ProfileResourceDELETE001 - Delete deviceResource by name
+    Given Set Test Variable  ${profile_name}  Test-Profile-1
+    And Generate A Device Profile Sample  ${profile_name}
+    And Create device profile ${deviceProfile}
+    When Delete Unused deviceResource with by Name
+    Then Should Return Status Code "200"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    And Device Resource ${resource_name} Should Be Deleted
+    [Teardown]  Delete device profile by name  ${profile_name}
+
+ErrProfileResourceDELETE001 - Delete deviceResource by non-existent deviceResource name
+    # non-existent deviceResource name
+    Given Set Test Variable  ${profile_name}  Test-Profile-1
+    And Generate A Device Profile Sample  ${profile_name}
+    And Create device profile ${deviceProfile}
+    When Delete deviceResource by Name not-existent in ${profile_name}
+    Then Should Return Status Code "404"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete Device Profile By Name  ${profile_name}
+
+ErrProfileResourceDELETE002 - Delete deviceResource which profile used by device
+    Given Set Test Variable  ${profile_name}  Test-Profile-2
+    And Create A Device Sample With Associated Test-Device-Service And ${profile_name}
+    When Delete Unused deviceResource with by Name
+    Then Should Return Status Code "409"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    And No Device Resource ${resource_name} Should Be Deleted
+    [Teardown]  Run Keywords  Delete Device By Name Test-Device
+    ...                  AND  Delete Device Service By Name  Test-Device-Service
+    ...                  AND  Delete Device Profile By Name  ${profile_name}
+
+ErrProfileResourceDELETE003 - Delete deviceResource which used by deviceCommand
+    Given Set Test Variable  ${profile_name}  Test-Profile-3
+    And Generate A Device Profile Sample  ${profile_name}
+    And Create device profile ${deviceProfile}
+    When Delete Used deviceResource with by Name
+    Then Should Return Status Code "400"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    And No Device Resource ${resource_name} Should Be Deleted
+    [Teardown]  Delete Device Profile By Name  ${profile_name}
+
+ErrProfileResourceDELETE004 - Delete deviceResource when StrictDeviceProfileChanges config is enabled
+    # StrictDeviceProfileChanges config is true
+    ${path}  Set Variable  /v1/kv/edgex/core/${CONSUL_CONFIG_VERSION}/core-metadata/Writable/ProfileChange/StrictDeviceProfileChanges
+    Given Update Service Configuration On Consul  ${path}  true
+    And Sleep  1s  # Waiting for the configuration updating
```

Comment:
Here is the keyword for StrictDeviceProfileChanges setting
https://github.com/edgexfoundry/edgex-taf/blob/279746c4e7ace1f145c2d5a5232eb71ca2f1b5d4/TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot#L618

Reply:
Updated.